src/snowcli/cli/nativeapp/manager.py

```diff
@@ -443,23 +501,44 @@ def app_run(self) -> None:
                     f"""
                     create application package {self.package_name}
                         comment = {SPECIAL_COMMENT}
-                        distribution = internal
+                        distribution = {self.package_distribution}
                     """
                 )
             else:
-                row_comment = show_app_row[
-                    COMMENT_COL
-                ]  # Because we use a DictCursor, we are guaranteed a dictionary instead of indexing through a list.
-
-                if row_comment != SPECIAL_COMMENT:
-                    raise ApplicationPackageAlreadyExistsError(self.package_name)
-
+                # 1. Check for the right owner role
                 actual_owner = show_app_row[OWNER_COL]
                 if actual_owner != unquote_identifier(self.package_role):
                     raise UnexpectedOwnerError(
                         self.app_name, self.app_role, actual_owner
                     )
 
+                # 2. Check distribution of the existing app package
+                actual_distribution = self.get_app_pkg_distribution_in_snowflake
+                if self.is_app_pkg_distribution_mismatched():
+                    log.warning(
+                        dedent(
+                            f"""\
+                        Continuing to execute `snow app run` on app pkg {self.package_name} with distribution '{actual_distribution}'.
```

Comment:
Consider being more explicit about what we're warning the user about, this doesn't indicate that there is a mismatch.